core/tee/tee_svc_storage.c

```diff
@@ -937,6 +937,8 @@ TEE_Result syscall_storage_obj_trunc(unsigned long obj, size_t len)
 			res = TEE_ERROR_GENERIC;
 	}
 
+	o->info.dataSize = len;
```

Comment:
````
	if (res == TEE_SUCCESS)
		o->info.dataSize = len;

Reply:
I missed that one, good catch!

Reply:
Oops, yeah great catch! I decided that making the error case above it early exit would be a bit cleaner.